pkg/executor/statefulset.go

```diff
@@ -679,19 +680,60 @@ func getStartupScript(
 	}
 }
 
+
+// Checks if the rsync command is available. 
+// If rsync is installed the RSYNC_INSTALLED variable will be 0.
+func genrateRsyncInstalledCmd() string {
+    cmd := "rsync -h &>/dev/null; RSYNC_INSTALLED=$?; "
+    return cmd;
+}
+
+// Generates command that will do copying with rsync
+// The progress will be stored in a file.
+func generateRsyncCmd(persistDirs []string) string {
+    
+    // The directory should be created in /mnt in advance,
+    // otherwise the rsync log file will not be created
+    createRsyncLogFileBaseDir := fmt.Sprintf("mkdir -p /mnt%s", filepath.Dir(kubedirectorInitProgressLogs))
+
+    rsyncCmd := fmt.Sprintf("%s; rsync --log-file=/mnt%s --relative -ax %s /mnt;",
+        createRsyncLogFileBaseDir,
+        kubedirectorInitProgressLogs, 
+        strings.Join(persistDirs, " "))
+
+    return rsyncCmd
+}
+
+// Generates command that will do copying with cp
+// No way to display progress
+func generateCpCmd(persistDirs []string) string {
```

Comment:
Same things here about function arguments and also the initial blank line.

Reply:
Done!